openpype/pipeline/workfile/path_resolving.py

```diff
@@ -534,3 +539,49 @@ def create_workdir_extra_folders(
         fullpath = os.path.join(workdir, subfolder)
         if not os.path.exists(fullpath):
             os.makedirs(fullpath)
+
+
+def get_last_workfile_representation(
+    project_name,
+    asset_name,
+    task_name,
+    host_name=None,
+    fields=None,
+):
+    """Get last published workfile representation.
+
+    Args:
+        project_name(str): Project name.
+        asset_name(str): Asset/Shot name.
+        task_name (str): Task name.
+        host_name (Optional[str]): Host name.
+        fields (Optional[Iterable[str]]): Fields that should be returned.
+            Defaults to None.
+
+    Returns:
+        dict: Last workfile representation.
+    """
+    # Get workfile extensions
+    host_module = ModulesManager().get_host_module(host_name)
```

Comment:
What's the advantage of using the `ModuleManager` versus `extensions = HOST_WORKFILE_EXTENSIONS.get(host_name)` directly?

In case that `host_name` is `None`, we would end up with either an error or an empty `extensions`, no? (as in we are not checking that `host_name is not None`)

Reply:
This is how extensions are determined in Applications. The way I understand it, `HOST_WORKFILE_EXTENSIONS` seems to be a fallback in case host module is not found. I don't think it should work differently here or else it could cause unexpected behavior if extensions have been edited in one place and not the other.

Also good point on `host_name`. In theory the method would return None in this case. But I think my mistake is to leave it as an optional argument. It should be mandatory now.